### PR TITLE
updated the filter code to include a cache of checked patterns

### PR DESCRIPTION
**Pull request by rlaquale**
I was using the antpath filter on large responses > 10MB.  In profiling the code it appeared that a large portion of time was being spent in matching.  I noticed that it appeared that there was a fair amount of repetitive matching against the same "nested paths."  So I thought it best to include a simple cache to ensure that matching wasn't done unnecessarily.  This has drastically reduced total filtering times for large responses.

### CODE REVIEW

**Comment by Antibrumm** [comment]
Thanks. I like the idea. I have not yet used it in large lists and i believe that this will help quite a bit. Did you do some benchmarks with an outcome of magnitude? Time spent or similar? Would be interesting. 

I will take a look at it soon.

**Comment by rlaquale** [comment]
Hi -- yes, we are currently profiling the change now.  I've got some results from a unit test that runs the filtering code 7 times against the same 9.5MB file, with progressively deeper paths.  We are using YourKit, and both runs are completely equal except for the addition of the caching code.

There is quite a reduction in the number of times the matching code it called.

no-cache
with cache

File: src/main/java/ch/mfrey/jackson/antpathfilter/AntPathPropertyFilter.java [hunk 1]
```diff
@@ -107,40 +118,90 @@ protected boolean include(final PropertyWriter writer) {
...
-     * @param jgen
-     *            the jgen
+     * @param writer the writer
+     * @param jgen the jgen
      * @return true, if successful
      */
     protected boolean include(final PropertyWriter writer, final JsonGenerator jgen) {
         String pathToTest = getPathToTest(writer, jgen);
         if (_propertiesToInclude.isEmpty()) {
+
```
Comment (Antibrumm) [inline src/main/java/ch/mfrey/jackson/antpathfilter/AntPathPropertyFilter.java:128]: In fact I think we can further optimize this cache implementation by completely avoiding the the "for loops". I wonder if you like to do it yourself and if you have time doing it. Else i can take over from here. The idea is great already :)
If you like to do it it's pretty simple:
- after `pathToTest` is known check directly the cache
- if cache found return result (true, false)
- else do the standard calculation + adding the result into the cache just before returning it.

Like this we can avoid all the for loops

**Comment by Antibrumm** [comment]
I have gladly accepted your pull request. I had some time, so I optimized the code a little. 
In fact, once we evaluated the `pathToTest` we can just capture the result and check this upfront. No need for any loops. Hope this shows an even bigger boost in your tests :)

**Comment by rlaquale** [comment]
Thank you!  The optimization looks great.  Look forward to running some tests against it. Thanks for the prompt merge and update.

**Pull request merged**